src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,75 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserNotValidException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    static final int MIN_LOGIN_LENGTH = 6;
+    static final int MIN_PASSWORD_LENGTH = 6;
+    static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        storageDao.add(user);
+        return storageDao.get(user.getLogin());
+    }
+
+    private void validateUser(User user) {
+        if (user == null) {
+            throw new UserNotValidException("User cant be null");
+        }
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+    }
+
+    private void validateLogin(User user) {
+        String userLogin = user.getLogin();
+
+        if (userLogin == null) {
+            throw new UserNotValidException("Login of user cant be null");
+        }
+        if (storageDao.get(userLogin) != null) {
+            throw new UserNotValidException(
+                    "User with this login exist already. Login is + " + user.getLogin()
+            );
+        }
```

Comment:
better to do this check after validation `serLogin.length()` cause you check all input data that is not null, but why should you do it if the password cannot be less than 6 characters?